### PR TITLE
`ForwardRequest` policy - added optional attribute builder - allow to set `follow-redirects`, `buffer-request-body`, `buffer-response`, `fail-on-error-status-code`, `timeout` optional attributes

### PR DESCRIPTION
**Pull request by cermakp**
`ForwardRequest` policy - added optional attribute builder - allow to set `follow-redirects`, `buffer-request-body`, `buffer-response`, `fail-on-error-status-code` optional attributes

### CODE REVIEW

File: tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs [hunk 1]
```diff
@@ -8,22 +8,12 @@ namespace Oriflame.PolicyBuilder.Xml.Tests.Definitions.Common
     public class ForwardRequestsTests
     {
         [Theory]
-        [InlineData(5)]
-        public void CreatesCorrectPolicy(int timeoutSeconds)
+        [InlineData("5")]
+        public void CreatesCorrectPolicy(string timeoutSeconds)
```
Comment (oskrabanek) [inline tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs]: Parameter and `InlineData` attribute not needed anymore.

File: tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs [hunk 2]
```diff
@@ -8,22 +8,12 @@ namespace Oriflame.PolicyBuilder.Xml.Tests.Definitions.Common
...
-            var basePolicy = new ForwardRequest(TimeSpan.FromSeconds(timeoutSeconds));
+            var basePolicy = new ForwardRequest(new Dictionary<string, string>());
             var xml = basePolicy.GetXml().ToString();
-            xml.Should().Be($"<forward-request timeout=\"{timeoutSeconds}\" />");
-        }
-        
-        [Theory]
-        [InlineData(5, true)]
-        [InlineData(3, false)]
-        public void CreatesCorrectBufferResponsePolicy(int timeoutSeconds, bool bufferResponse)
```
Comment (oskrabanek) [inline tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs:22]: I would leave this test as a partial one, just update the body to use the AttributeBuilder.

**Review by oskrabanek — CHANGES_REQUESTED** [review]

File: tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs (hunk 2 above)
Comment (cermakp) [reply tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs:22]: This is already covered by `BackendSectionPolicyBuilderTests`. It does not make sense to cover here the attribute logic builder logic. So the only think I can test here is to pass some IDictionary collection of attributes but not sure if we want to do it here, I prefer to test it in `BackendSectionPolicyBuilderTests`.

File: tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs (hunk 1 above)
Comment (cermakp) [reply tests/Oriflame.PolicyBuilder.Xml.Tests/Definitions/Common/ForwardRequestsTests.cs]: You are right, removed

**Review by oskrabanek — APPROVED** [review]

**Pull request merged**